### PR TITLE
Firearms subpage and searchbar

### PR DESCRIPTION
**Pull request by jordykoppen**
* Added Firearms page component + route in stackrouter
* Added Small Card for items in subpage
* Added Component for Horizontal scrolling (Small Cards)
* Added Searchbar to Homescreen. Awaiting implementation with backend/firebase

The Firearms page will be generic once the other data comes in. So we don't have to create a separate component for each "Category" unless necessary.

I wan't to make the `Card` and `SmallCard` into one component but waiting for feedback. Will fix in coming PR's.

### CODE REVIEW

File: android/battleBuddyAndroid/components/common/Card.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ import styled, { css } from 'styled-components/native';
 import PropTypes from 'prop-types';
 
 const StyledCard = styled.ImageBackground`
+  /* prettier-ignore */
```
Comment (jordykoppen) [inline android/battleBuddyAndroid/components/common/Card.js:6]: Ignore the ignore comments for now :P 
It's an issue with one of my linters/formatters that keep converting the camelCase names to lowercase. Will look into a fix once I integrate prettier and eslint

**Comment by jordykoppen** [comment]
Last 3 commits are for eslint, prettier and lint-staged integration. Accidentally added it to same branch ¯\_(ツ)_/¯

Don't forget to run `npm install` after merging

**Comment by VeritasDev** [comment]
LMK when this is ready to review fully

**Comment by jordykoppen** [comment]
@VeritasDev Ready :)

**Pull request merged**